tests/test_contracts.nim

```diff
@@ -157,7 +157,7 @@ suite "Contracts":
     waitFor asyncteardown()
 
   test "encoding test":
-    proc asynctest {.async.} =
+    proc asynctest {.async: (raises: [CancelledError, Exception]).} =
```

Comment:
what's the source of the `Exception`? we shoulnd't have them leaking from anywhere really